src/ui/public/saved_objects/saved_objects_client.js

```diff
@@ -0,0 +1,95 @@
+import { resolve as resolveUrl, format as formatUrl } from 'url';
+import { pick, partial } from 'lodash';
+
+import { SavedObject } from './saved_object';
+
+const join = (...uriComponents) => (
+  uriComponents.filter(Boolean).map(encodeURIComponent).join('/')
+);
+
+export class SavedObjectsClient {
+  constructor($http, basePath) {
+    this._$http = $http;
+    this._apiBaseUrl = `${basePath}/api/saved_objects/`;
+  }
+
+  get(type, id) {
+    if (!type || !id) {
+      return Promise.reject(new Error('requires type and id'));
```

Comment:
Using the global `Promise` constructor here might cause problems with angular's digest cycle.

One of these is true:
 1. angular will not trigger a digest cycle when the non-angular promise is resolved/rejected and the changes that occur as a result will not be picked up by the dirty checks properly and the UI will hang until some other action occurs (like a click in the general area of the affected directive).
 1. angular will consume the non-angular promise and convert it to an angular promise when it encounters it, or the Promise resolving/rejecting won't properly trigger a digest cycle like it should.

I'm pretty sure that option 1 is true. I proved it in the past because we were having very strange lags/snags in the UI that required arbitrary clicks to trigger digest cycles so the UI would become responsive.

Hopefully though, option 2 is true. If it is then there is no reason these can't use `async/await`, and we could `throw new Error('requires type and id')` rather than explicitly returning a rejected promise.